src/Raven.Client/Documents/Session/AsyncDocumentSession.cs

```diff
@@ -64,7 +66,28 @@ public async Task RefreshAsync<T>(T entity, CancellationToken token = default(Ca
                 var command = new GetDocumentsCommand(new[] { documentInfo.Id }, includes: null, metadataOnly: false);
                 await RequestExecutor.ExecuteAsync(command, Context, _sessionInfo, token).ConfigureAwait(false);
 
-                RefreshInternal(entity, command, documentInfo);
+                var commandResult = (BlittableJsonReaderObject)command.Result.Results[0];
+                RefreshInternal(entity, commandResult, documentInfo);
+            }
+        }
+
+        public async Task RefreshAsync<T>(T[] entities, CancellationToken token = default(CancellationToken))
+        {
```

Comment:
I think you missed `using (AsyncTaskHolder())` ?

Reply:
Done